finesse/gui/led_icons.py

```diff
@@ -0,0 +1,68 @@
+"""Class for LED Icons."""
+from importlib import resources
+
+from PySide6.QtCore import QTimer
+from PySide6.QtGui import QImage, QPixmap
+from PySide6.QtWidgets import QLabel
+
+_img_files = resources.files("finesse.gui.images")
+_poll_on_img_data = _img_files.joinpath("poll_on.png").read_bytes()
+_poll_off_img_data = _img_files.joinpath("poll_off.png").read_bytes()
+_alarm_on_img_data = _img_files.joinpath("alarm_on.png").read_bytes()
+_alarm_off_img_data = _img_files.joinpath("alarm_off.png").read_bytes()
+
+_poll_on_img = QImage.fromData(_poll_on_img_data)
+_poll_off_img = QImage.fromData(_poll_off_img_data)
+_alarm_on_img = QImage.fromData(_alarm_on_img_data)
+_alarm_off_img = QImage.fromData(_alarm_off_img_data)
+
+
+class LEDIcon(QLabel):
+    """QLabel object to represent an LED with on/off status."""
+
+    def __init__(self, on_img: QImage, off_img: QImage, is_on: bool = False) -> None:
+        """Creates the LED icon, sets its status and stores corresponding image data.
+
+        Args:
+            on_img (QImage): QImage for LED on state.
+            off_img (QImage): QImage for LED off state.
+            is_on (bool): On/off status of LED.
+        """
+        super().__init__()
+        self._on_img = on_img
+        self._off_img = off_img
+        if is_on:
+            self._turn_on()
+        else:
+            self._turn_off()
+        self._timer = QTimer()
+        self._timer.timeout.connect(self._turn_off)  # type: ignore
+
+    @staticmethod
+    def create_poll_icon() -> "LEDIcon":
+        """Creates the LED icon for polling the server."""
+        return LEDIcon(on_img=_poll_on_img, off_img=_poll_off_img)
+
+    @staticmethod
+    def create_alarm_icon() -> "LEDIcon":
+        """Creates the LED icon to indicate alarm status."""
+        return LEDIcon(on_img=_alarm_on_img, off_img=_alarm_off_img)
```

Comment:
The usual approach when dealing with factory methods like these is:
```suggestion
    @clasmethod
    def create_poll_icon(cls) -> LEDIcon:
        """Creates the LED icon for polling the server."""
        return cls(on_img=_poll_on_img, off_img=_poll_off_img)

    @clasmethod
    def create_alarm_icon(cls) -> LEDIcon:
        """Creates the LED icon to indicate alarm status."""
        return cls(on_img=_alarm_on_img, off_img=_alarm_off_img)
```

For this to work, you will need to import at the top of the file:

```python
from __futures__ import annotations
```